cli/cli.go

```diff
@@ -26,17 +26,27 @@ func Run() {
 			Usage:  "debug mode",
 			EnvVar: "DEBUG",
 		},
-
 		cli.StringFlag{
 			Name:  "log-level, l",
 			Value: "info",
 			Usage: fmt.Sprintf("Log level (options: debug, info, warn, error, fatal, panic)"),
 		},
+		cli.StringFlag{
+			Name:  "log-file",
+			Usage: "Logging the output to a file instead of stdout",
+		},
 	}
-
 	// logs
 	app.Before = func(c *cli.Context) error {
-		log.SetOutput(os.Stderr)
+		if c.IsSet("log-file") {
+			logDir := c.String("log-file")
+			fmt.Print(logDir)
+			file, err := os.OpenFile(logDir, os.O_RDWR, 0777)
```

Comment:
We should check for the error here opening the file, if it fails: try to create the file and the directory hierarchy. If there is a failure creating the file or directory, then fall-back to default log mode (with `os.Stderr`).